server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -160,6 +162,7 @@ public TransportSearchAction(
         this.namedWriteableRegistry = namedWriteableRegistry;
         this.executorSelector = executorSelector;
         this.defaultPreFilterShardSize = DEFAULT_PRE_FILTER_SHARD_SIZE.get(clusterService.getSettings());
+        this.fetchLookupFieldsPhase = new FetchLookupFieldsPhase(searchTransportService, searchService::isAllowExpensiveQueries);
```

Comment:
If we changed the CCS behavior so the lookup index is resolved in the same cluster, could we convert this to a "standard" search phase class?